### PR TITLE
Print flops pruning level in statistic

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
Example of output:
```
pruning_rate: 0.09999999999999998
FLOPS pruning level: 0.1128178144173182
FLOPS current / full: 533683152 / 601548544
```

Additions fixes:
- Missed transpose convolution for 'prunable_types'
- Fix incorrect computation FLOPS of linear module

### CODE REVIEW

File: nncf/pruning/filter_pruning/algo.py [hunk 1]
```diff
@@ -166,6 +167,27 @@ def compute_cost_hook(module, input_, output):
         for h in hook_list:
             h.remove()
 
+    def _calculate_flops_pruned_model_by_masks(self):
```
Comment (mkaglins) [inline nncf/pruning/filter_pruning/algo.py:170]: Please, add any tests on this statistic calculation.

File: nncf/pruning/filter_pruning/algo.py [hunk 2]
```diff
@@ -90,9 +89,11 @@ def __init__(self, target_model: NNCFNetwork,
     def _get_mask(minfo: PrunedModuleInfo):
         return minfo.operand.binary_filter_pruning_mask
 
-    def statistics(self, quickly_collected_only=False):
-        stats = super().statistics()
+    def add_algo_specific_stats(self, stats):
         stats['pruning_rate'] = self.pruning_rate
+        flops = self._calculate_flops_pruned_model_by_masks()
```
Comment (mkaglins) [inline nncf/pruning/filter_pruning/algo.py:94]: Here you can re-use pruning rate as FLOPs in case of `self.prune_flops == True`

**Review by mkaglins — CHANGES_REQUESTED** [review]

**Comment by AlexanderDokuchaev** [comment]
Jenkins please retry a build

File: nncf/utils.py [hunk 3]
```diff
@@ -369,7 +369,10 @@ def compute_FLOPs_hook(module, input_, output, dict_to_save, name):
         ks = module.weight.data.shape
         mac_count = np.prod(ks) * np.prod(output.shape[2:])
     elif isinstance(module, nn.Linear):
-        mac_count = input_[0].shape[1] * output.shape[-1]
+        if len(input_[0].shape) == 1:
```
Comment (AlexKoff88) [inline nncf/utils.py:372]: Please add a comment that explains the need for this change.

**Comment by AlexKoff88** [comment]
Can you please provide an example of new output?

**Review by AlexKoff88 — COMMENTED** [review]
Please resolve the comments from Maria.

**Comment by AlexanderDokuchaev** [comment]
@AlexKoff88 add example of output in PR comment.

**Pull request merged**